event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/resource/EventStreamPageResource.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.services.eventsourcing.source.api.resource;
+
+
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.BACKWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.FORWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.valueOf;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.FIRST;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.HEAD;
+
+import uk.gov.justice.services.common.converter.ObjectToJsonValueConverter;
+import uk.gov.justice.services.common.converter.jackson.ObjectMapperProducer;
+import uk.gov.justice.services.eventsourcing.source.api.feed.common.Page;
+import uk.gov.justice.services.eventsourcing.source.api.security.AccessController;
+import uk.gov.justice.services.eventsourcing.source.api.service.EventStreamPageService;
+
+import java.net.MalformedURLException;
+import java.sql.SQLException;
+
+import javax.inject.Inject;
+import javax.json.JsonValue;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.UriInfo;
+
+@Path("event-streams/{position}/{direction}/{pageSize}")
+public class EventStreamPageResource {
+
+    @Context
+    HttpHeaders headers;
+
+    @Inject
+    private EventStreamPageService eventsPageService;
+
+    @Inject
+    private AccessController accessController;
+
+    @Inject
+    private ObjectMapperProducer mapperProducer;
+
+    @Inject
+    private ObjectToJsonValueConverter converter;
+
+    @GET
+    @Produces("application/vnd.event-source.events+json")
+    public JsonValue events(
+            @PathParam("position") final String position,
+            @PathParam("direction") final String direction,
+            @PathParam("pageSize") final long pageSize,
+            @Context final UriInfo uriInfo) throws SQLException, MalformedURLException {
+
+        if (HEAD.getPosition().equals(position) && FORWARD.toString().equals(direction)) {
```

Comment:
Thought mentioned before updating the error text on old review, Perhaps: Invalid request, cannot request PREVIOUS page from HEAD.

is it worth putting this into a small util that we can reuse across the two?

Or maybe a private method. validateRequest()

Reply:
moved to a private method